daemon/access.go

```diff
@@ -255,3 +263,50 @@ func (ac interfaceAuthenticatedAccess) CheckAccess(d *Daemon, r *http.Request, u
 
 	return Unauthorized("access denied")
 }
+
+// isRequestFromSnapCmd checks that the request is coming from snap command.
+//
+// It checks that the request process "/proc/PID/exe" points to one of the
+// known locations of the snap command.
+func isRequestFromSnapCmd(st *state.State, r *http.Request) (bool, error) {
+	ucred, err := ucrednetGet(r.RemoteAddr)
+	if err != nil {
+		return false, err
+	}
+	exe, err := osReadlink(fmt.Sprintf("/proc/%d/exe", ucred.Pid))
+	if err != nil {
+		return false, err
+	}
+
+	// SNAP_REEXEC=0
+	if exe == filepath.Join(dirs.GlobalRootDir, "/usr/bin/snap") {
+		return true, nil
+	}
+
+	// Check if re-exec in snapd
+	var snapst snapstate.SnapState
+	err = snapstate.Get(st, "snapd", &snapst)
+	if err != nil && !errors.Is(err, state.ErrNoState) {
+		return false, err
+	}
+	if err == nil {
+		snapdPath := fmt.Sprintf("snapd/%s/usr/bin/snap", snapst.CurrentSideInfo().Revision)
```

Comment:
does it have to be current, or would any installed revision be acceptable? i.e. what if snapd is refreshing just now.

Reply:
updated to use wildcard matching, great catch, thanks!